app/controllers/publishers/omniauth_callbacks_controller.rb

```diff
@@ -157,6 +157,31 @@ def register_twitter_channel
       return
     end
 
+    def register_vimeo_channel
+      vimeo_auth_hash = request.env['omniauth.auth']
+      @channel = current_publisher.channels.new(verified: true)
+      @channel.details = VimeoChannelDetails.new(
+        name: vimeo_auth_hash.info.name,
+        vimeo_channel_id: vimeo_auth_hash.info.id,
+        auth_provider: vimeo_auth_hash.info.auth_provider,
+        thumbnail_url: vimeo_auth_hash.info.pictures.last.link,
+        channel_url: vimeo_auth_hash.info.link,
+        nickname: vimeo_auth_hash.info.nickname
+      )
+
+      existing_channel = Channel.joins(:vimeo_channel_details).where("vimeo_channel_details.vimeo_channel_id": vimeo_auth_hash.info.id).first
```

Comment:
Let's make sure the existing vimeo channel that we're trying to find is verified